File: minimal-go/minimal.go
package main

import (
	"fmt"
	"os"
	"strings"
	"syscall"
)

func main() {
	fmt.Println("This is MINIMAL (in GO)")

	fmt.Println()
	fmt.Println("Arguments:")
	for _, a := range os.Args {
		fmt.Println("- " + "'" + a + "'")
	}

	fmt.Println()
	fmt.Println("Environment:")
	for _, e := range os.Environ() {
		pair := strings.Split(e, "=")
		fmt.Println("- " + pair[0] + "='" + pair[1] + "'")
	}
	
	fmt.Printf("User: %d:%d\n", syscall.Getuid(), syscall.Getgid())	
}

File: minimal-c/minimal.c
#include <stdio.h>
#include <unistd.h>

int main(int argc, char** argv) {
    puts("This is MINIMAL (in C)");

    puts("");
    printf("Arguments (count: %i):\n", argc);
    for(int i = 0; i < argc; i++) {
        printf("- '%s'\n", argv[i]);
    }

    puts("");
    printf("Environment:\n");
    char** current = environ;
    while(*current) {
        printf("- '%s'\n", *current);
        current++;
    }

    puts("");
    printf("User: real %i:%i, effective %i:%i\n", getuid(), getgid(), geteuid(), getegid());
}
